### PR TITLE
Update connection.js

### PR DESCRIPTION
**Pull request by iamdanthedev**
Hello

We have custom extenstions to Array.prototype in our project
`for .. in` cycles through all of them and of course crashes
Either `hasOwnProperty` should be used in order to filter out the other keys or `for .. of` works perfectly fine and is more consice

Thanks

### CODE REVIEW

File: lib/connection.js [hunk 1]
```diff
@@ -230,6 +230,10 @@ var Connection = function (options, container) {
     this.closed_with_non_fatal_error = false;
 
     for (var i in aliases) {
+        if (!aliases.hasOwnProperty(i)) {
+            continue;
+        }
+
```
Comment (grs) [inline lib/connection.js]: Though it doesn't show up in the github ui here, you have four spaces on this blank line which is failing the lint test.

File: lib/connection.js [hunk 2]
```diff
@@ -230,6 +230,10 @@ var Connection = function (options, container) {
     this.closed_with_non_fatal_error = false;
 
     for (var i in aliases) {
```
Comment (grs) [inline lib/connection.js]: Perhaps aliases.forEach() might be cleaner here?

Comment (iamdanthedev) [reply lib/connection.js]: yeah, and there is no need to check for hasOwnProperty in this case
I'll send an update now

**Comment by iamdanthedev** [comment]
Hi @grs 
I've updated the PR, could you check if can be merged?
I didn't realise that fat arrows are supported all the way back to node4

There are other similar places in the code which could be a potential danger, but don't seem to crash node at the moment so I didn't take a close look at them:

image

But please let me know if they need to be updated as well

**Comment by grs** [comment]
Thanks @iamdanthedev !

**Comment by iamdanthedev** [comment]
Thanks! But how to get it to npm?

**Comment by grs** [comment]
Released as 1.0.19

**Comment by iamdanthedev** [comment]
thanks!

**Pull request merged**